x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/RoleDescriptorRequestValidator.java

```diff
@@ -71,6 +78,11 @@ public static ActionRequestValidationException validate(
             } catch (IllegalArgumentException ile) {
                 validationException = addValidationError(ile.getMessage(), validationException);
             }
+            if (DataStream.isFailureStoreFeatureFlagEnabled()) {
+                for (String indexName : ridp.indicesPrivileges().getIndices()) {
+                    validationException = validateIndexNameExpression(indexName, validationException);
```

Comment:
We don't support failure store privileges yet for RCS2, but I think we should also prevent using `::` in remote index patterns as well.